### ISSUE TITLE
Switch "Turn on/off TBH" on C3 type device - Heat Pump Wi-Fi Controller

### ISSUE THREAD

**Issue by erikgieseler** [georgezhao2010/midea_ac_lan#345]
### Device type and model (or SN)

C3 - Heat Pump Wi-Fi Controller 171H120F

### The description of new feature

Hello,

could you please also maybe add the switch "Turn on/off TBH"? 

I have attached a screenshot of the app and a debug log where I turned the TBH on and off via the app.

TBH on: 2023-10-07 13:30:01.811
TBH off: 2023-10-07 13:31:42.289

Thank you very much for your good work!

[home-assistant_midea_ac_lan_2023-10-07T11-34-24.301Z.log](https://github.com/georgezhao2010/midea_ac_lan/files/12837625/home-assistant_midea_ac_lan_2023-10-07T11-34-24.301Z.log)

![Screenshot_20231007-132110](https://github.com/georgezhao2010/midea_ac_lan/assets/72557610/6576a3c9-27bf-408f-9845-5a60e2b97443)

**Comment by georgezhao2010** [comment]
> ### Device type and model (or SN)
> C3 - Heat Pump Wi-Fi Controller 171H120F
> 
> ### The description of new feature
> Hello,
> 
> could you please also maybe add the switch "Turn on/off TBH"?
> 
> I have attached a screenshot of the app and a debug log where I turned the TBH on and off via the app.
> 
> TBH on: 2023-10-07 13:30:01.811 TBH off: 2023-10-07 13:31:42.289
> 
> Thank you very much for your good work!
> 
> [home-assistant_midea_ac_lan_2023-10-07T11-34-24.301Z.log](https://github.com/georgezhao2010/midea_ac_lan/files/12837625/home-assistant_midea_ac_lan_2023-10-07T11-34-24.301Z.log)
> 

Got it, thank you for your logs.

**Comment by georgezhao2010** [comment]
Hello,
The TBH function seems to be controlled by the switch disinfect. Could you try and see if it works?

```javascript
buttons: {
	cancel: "Cancel",
	ok: "OK",
	confirm: "Confirm",
	save: "Save",
	del: "Delete",
	ex_mode: "Mode",
	ex_curve: "Curve",
	ex_dis_infect: "Turn on TBH.",
	ex_fast_dhw: "Fast DHW",
	fun_schedule: "Schedule",
	fun_slient: "Silent",
	fun_holiday: "Holiday",
	fun_eco: "ECO"
},
```

**Comment by georgezhao2010** [comment]
> ### The description of new feature
> Hello,
> 
> could you please also maybe add the switch "Turn on/off TBH"?
> 
> I have attached a screenshot of the app and a debug log where I turned the TBH on and off via the app.
> 
> TBH on: 2023-10-07 13:30:01.811 TBH off: 2023-10-07 13:31:42.289
> 
> Thank you very much for your good work!

Hello,
The TBH function seems to be controlled by the switch disinfect. Could you try and see if it works?

```javascript
buttons: {
	cancel: "Cancel",
	ok: "OK",
	confirm: "Confirm",
	save: "Save",
	del: "Delete",
	ex_mode: "Mode",
	ex_curve: "Curve",
	ex_dis_infect: "Turn on TBH.",
	ex_fast_dhw: "Fast DHW",
	fun_schedule: "Schedule",
	fun_slient: "Silent",
	fun_holiday: "Holiday",
	fun_eco: "ECO"
},
```

**Comment by erikgieseler** [comment]
Correct, when I activate the switch disinfect in HA, the TBH (in German = Tankheizer) turns on, see attached pictures. In the app, "Turn TBH on" is then also displayed as active. The "TBH status" in HA then also changes correctly. The heat pump also has a separate disinfection mode which is not activated when you activate the switch disinfect in HA. The name of the switch "disinfect" would be wrong in this case because the disinfect mode is not activated. You could simply rename or copy the disinfect switch in this case.

![PXL_20231010_151259912](https://github.com/georgezhao2010/midea_ac_lan/assets/72557610/6bb76d6f-1b14-43ff-974e-ed9672aa7a0f)
![PXL_20231010_151557599](https://github.com/georgezhao2010/midea_ac_lan/assets/72557610/c0550fe9-d07f-4bea-8a69-a08eb99398bc)
![PXL_20231010_151251825](https://github.com/georgezhao2010/midea_ac_lan/assets/72557610/c57c52f7-1175-44d4-9310-2c9a13f5293d)

**Comment by georgezhao2010** [comment]
> Correct, when I activate the switch disinfect in HA, the TBH (in German = Tankheizer) turns on, see attached pictures. In the app, "Turn TBH on" is then also displayed as active. The "TBH status" in HA then also changes correctly. The heat pump also has a separate disinfection mode which is not activated when you activate the switch disinfect in HA. The name of the switch "disinfect" would be wrong in this case because the disinfect mode is not activated. You could simply rename or copy the disinfect switch in this case.
> 

OK, The original code in the Midea app looks chaotic, and I’m not very clear about the logic of each function, so, I will just simply copy the disinfect switch as tbh.

**Comment by georgezhao2010** [comment]
implemented in v0.3.22
